### DISCUSSION TITLE
Are checksum crate names allowed?

### DISCUSSION THREAD

**Discussion by heisen-li** [rust-lang/crates.io#4227] [Q&A] [answered]
Are there currently similar community norms? Or check mechanism? Is this behavior allowed?

![image](https://user-images.githubusercontent.com/46313511/144345298-abcd656e-765b-4b20-90b0-51f74f8ae83d.png)

**Comment by Turbo87** [comment, marked as answer]
AFAICT it is not against our current publishing policies, though I admit that I'm not a huge fan of this behavior either. As we've written here multiple times already, changing the policies will most likely require an RFC. You're very welcome to work on that if you're interested in helping.

**Reply by heisen-li** [reply]
Of course, I'm more than happy to help. This behavior is obviously inappropriate, but I haven't figured out how to prevent it. Or, what should I do next?

**Reply by carols10cents** [reply]
I think the first step is articulating *why* the behavior is obviously inappropriate (that is, how exactly it hurts the Rust community and how crates.io would automatically detect it). Then work on several proposed solutions and write out the advantages and disadvantages of each. Take a look at the RFC instructions in the [RFC repo's README](https://github.com/rust-lang/rfcs), and feel free to open a "pre-RFC" on https://internals.rust-lang.org/ to get early feedback.

**Reply by heisen-li** [reply]
Thanks Reply. I created the most original discussion thread. Many ideas are immature and need to get feedback from everyone. I will continue to follow this discussion. If you have better descriptions and suggestions, please feel free to enlighten me.https://internals.rust-lang.org/t/pre-rfc-prevent-abnormal-release-of-crates/15729
